bundles/action/org.openhab.action.telegram/src/main/java/org/openhab/action/telegram/internal/Telegram.java

```diff
@@ -130,4 +143,107 @@ static public boolean sendTelegram(@ParamDoc(name = "group") String group,
 
 		return sendTelegram(group, String.format(format, args));
 	}
+
+	@ActionDoc(text = "Sends a Picture via Telegram REST API")
+	static public boolean sendTelegramPhoto(
+			@ParamDoc(name = "group") String group,
+			@ParamDoc(name = "photoURL") String photoURL,
+			@ParamDoc(name = "caption") String caption) {
+
+		if (groupTokens.get(group) == null) {
+			logger.error("Bot '{}' not defined, action skipped", group);
+			return false;
+		}
+
+		if (photoURL == null) {
+			logger.error("photoURL not defined, action skipped");
+			return false;
+		}
+
+		// load image from url
+		byte[] imageFromURL;
+
+		HttpClient getClient = new HttpClient();
+		GetMethod getMethod = new GetMethod(photoURL);
+		getMethod.getParams().setParameter(HttpMethodParams.RETRY_HANDLER,
+				new DefaultHttpMethodRetryHandler(3, false));
+		try {
+			int statusCode = getClient.executeMethod(getMethod);
+			if (statusCode != HttpStatus.SC_OK) {
+				logger.error("Method failed: {}", getMethod.getStatusLine());
+				return false;
+			}
+
+			imageFromURL = getMethod.getResponseBody();
+		} catch (HttpException e) {
+			logger.error("Fatal protocol violation: {}", e.toString());
+			return false;
+		} catch (IOException e) {
+			logger.error("Fatal transport error: {}", e.toString());
+			return false;
+		} finally {
+			getMethod.releaseConnection();
+		}
+
+		// parse image type
+		String imageType;
+		try {
+			ImageInputStream iis = ImageIO
+					.createImageInputStream(new ByteArrayInputStream(
+							imageFromURL));
+			Iterator<ImageReader> imageReaders = ImageIO.getImageReaders(iis);
+			if (!imageReaders.hasNext()) {
+				logger.error("photoURL does not represent a known image type");
+				return false;
+			}
+			ImageReader reader = (ImageReader) imageReaders.next();
+			imageType = reader.getFormatName();
+		} catch (IOException e) {
+			logger.error("cannot parse photoURL as image: {}", e.getMessage());
+			return false;
+		}
+
+		// post photo to telegram
+		String url = String.format(TELEGRAM_PHOTO_URL, groupTokens.get(group)
+				.getToken());
+
+		PostMethod postMethod = new PostMethod(url);
+		try {
+			postMethod.getParams().setContentCharset("UTF-8");
+			postMethod.getParams().setSoTimeout(HTTP_TIMEOUT);
+			postMethod.getParams().setParameter(HttpMethodParams.RETRY_HANDLER,
+					new DefaultHttpMethodRetryHandler(3, false));
+			Part[] parts = new Part[caption != null ? 3 : 2];
+			parts[0] = new StringPart("chat_id", groupTokens.get(group)
+					.getChatId());
+			parts[1] = new FilePart("photo", new ByteArrayPartSource(
+					String.format("image.%s", imageType), imageFromURL));
```

Comment:
@watou 
If you are suggesting to get the name from the url, in my opinion this is not a good idea.
The url usually is something like http://..../get.php?date=11122015&height=24&width=250&res=high&....
In general the content type of the image is sent along the http headers and not in the name itself.

My code makes an analysis of the actual bytes received in order to guess the content type received, so that the telegram is happy with the file extension. And the name itself is not used by telegram apart from getting the extension, it is there just because required from the ByteArrayPartSource (if you use the FilePart instead of ByteArrayPartSource a name is not even required).
If you make a "save as" from telegram desktop the actual filename proposed is not the sent one but it is something like 83456634889_1232656545353543282341.jpg

So in my opinion my approach is fine.